### PR TITLE
add editorial book and sort

### PR DESCRIPTION
**Pull request by Riolku**
Change Editorial column header to a book icon, and implement editorial sorting. Also clean up Sum() clauses for editorial queries, since problems can have at most one editorial.

problems list

### CODE REVIEW

File: templates/problem/list.html [hunk 1]
```diff
@@ -233,7 +233,8 @@ <h3>{{ _('Hot problems') }} <i class="fa fa-fire"></i></h3>
                             <a href="{{ sort_links.ac_rate }}">{{ _('AC %%') }}{{ sort_order.ac_rate }}</a>
                         </th>
                         <th class="editorial">
-                            {{ _('Editorial') }}
+                            <a href="{{ sort_links.editorial }}"><i class="fa fa-book"></i>{{ sort_order.editorial }}
```
Comment (Ninjaclasher) [inline templates/problem/list.html:236]:
```suggestion
                            <a href="{{ sort_links.editorial }}">
                                <i class="fa fa-book"></i>{{ sort_order.editorial }}
```

File: judge/views/problem.py [hunk 2]
```diff
@@ -294,6 +294,13 @@ def get_paginator(self, queryset, per_page, orphans=0,
                 queryset = queryset.order_by(self.order.replace('name', 'i18n_name'), 'id')
             elif sort_key == 'group':
                 queryset = queryset.order_by(self.order + '__name', 'id')
+            elif sort_key == 'editorial':
+                queryset.annotate(has_public_editorial=Case(
```
Comment (Ninjaclasher) [inline judge/views/problem.py:298]: Is this annotation necessary? We already annotate unconditionally in `get_normal_queryset`.

File: templates/problem/list.html (hunk 1 above)
Comment (Riolku) [reply templates/problem/list.html]: Fixed, and also updated the formatting of the place a few lines above from which I copied it.

File: judge/views/problem.py (hunk 2 above)
Comment (Riolku) [reply judge/views/problem.py]: Fixed, good catch, thanks.

**Review by quantum5 — APPROVED** [review]

**Pull request merged**